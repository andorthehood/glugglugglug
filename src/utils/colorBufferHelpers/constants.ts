export const VERTICES_PER_QUAD = 6;
export const COLOR_COMPONENTS_PER_QUAD = 3;
