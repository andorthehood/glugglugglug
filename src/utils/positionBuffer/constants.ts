export const VERTICES_PER_QUAD = 6;
