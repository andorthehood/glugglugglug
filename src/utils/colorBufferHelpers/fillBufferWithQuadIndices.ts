import { VERTICES_PER_QUAD } from "./constants";

/**
 * Fills a buffer segment with the quad index repeated for each vertex.
 * @param buffer Destination Uint8Array storing quad indices per vertex.
 * @param quadIndex The quad index to write.
 * @param offset Starting index within the buffer.
 */
export function fillBufferWithQuadIndices(
	buffer: Uint8Array,
	quadIndex: number,
	offset: number,
): void {
	for (let i = 0; i < VERTICES_PER_QUAD; i += 1) {
		buffer[offset + i] = quadIndex;
	}
}
