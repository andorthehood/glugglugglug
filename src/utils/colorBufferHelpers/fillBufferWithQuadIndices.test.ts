import { describe, it, expect } from "vitest";

import { VERTICES_PER_QUAD } from "./constants";
import { fillBufferWithQuadIndices } from "./fillBufferWithQuadIndices";

function toArray(values: Uint8Array) {
	return Array.from(values);
}

describe("fillBufferWithQuadIndices", () => {
	it("fills a buffer segment with the same quad index", () => {
		const buffer = new Uint8Array(VERTICES_PER_QUAD);

		fillBufferWithQuadIndices(buffer, 2, 0);

		expect(toArray(buffer)).toMatchSnapshot();
	});

	it("respects the provided offset", () => {
		const buffer = new Uint8Array(VERTICES_PER_QUAD * 2);

		fillBufferWithQuadIndices(buffer, 5, VERTICES_PER_QUAD);

		expect(toArray(buffer)).toMatchSnapshot();
	});
});
