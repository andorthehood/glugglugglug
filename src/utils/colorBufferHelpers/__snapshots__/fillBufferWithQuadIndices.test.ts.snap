// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fillBufferWithQuadIndices > fills a buffer segment with the same quad index 1`] = `
[
  2,
  2,
  2,
  2,
  2,
  2,
]
`;

exports[`fillBufferWithQuadIndices > respects the provided offset 1`] = `
[
  0,
  0,
  0,
  0,
  0,
  0,
  5,
  5,
  5,
  5,
  5,
  5,
]
`;
